EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -0,0 +1,466 @@
+
+  void finalizeHeadTail(uhtrData* uhtr , bool verbosity){
+
+    uint64_t uhtr_size = uhtr->size()-8;
+
+    // adjust the size bits
+    uhtr->at(0) = uhtr_size&0xFFFF ;
+    uhtr->at(1) |= (uhtr_size>>16)&0xF ; 
+
+    // add trailer
+    uhtr->push_back( uhtr_size&0xFFFF );
+    uhtr->push_back( (uhtr_size>>16)&0xF );
+    // this is ignoring the event number... I am not sure what this should be
+
+    // adding some blank stuff for the CRC bits
+    uhtr->push_back( 0 );
+    uhtr->push_back( 0 );
+  };
+
+  void addChannel( int uhtrIndex , edm::SortedCollection<HFDataFrame>::const_iterator& qiedf , int verbosity = 0 ){
+    if( qiedf->size() == 0 ) return;
+    uint16_t header = packQIE8header(qiedf->sample(0), qiedf->elecId());
+    uhtrs[uhtrIndex].push_back(header);
+    // loop over words in dataframe
+    for(int iTS = 0; iTS < qiedf->size(); iTS +=2 ){
+       uint16_t cont =0;
+       int adc0 = qiedf->sample(iTS).adc();
+       int adc1 = qiedf->sample(iTS+1).adc();
+       cont |= adc0&0xFF;
+       cont |= (adc1&0xFF)<<8;
+       uhtrs[uhtrIndex].push_back(cont);
+    }// end loop over dataframe words
+  };
+
+  void addChannel( int uhtrIndex , edm::SortedCollection<HBHEDataFrame>::const_iterator qiedf , int verbosity = 0 ){
+    if( qiedf->size() == 0 ) return;
+    uint16_t header = packQIE8header(qiedf->sample(0), qiedf->elecId());
+    uhtrs[uhtrIndex].push_back(header);
+    // loop over words in dataframe
+    for(int iTS = 0; iTS < qiedf->size(); iTS +=2 ){
+       uint16_t cont =0;
+       int adc0 = qiedf->sample(iTS).adc();
+       int adc1 = qiedf->sample(iTS+1).adc();
+       cont |= adc0&0xFF;
+       cont |= (adc1&0xFF)<<8;
+       uhtrs[uhtrIndex].push_back(cont);
+    }// end loop over dataframe words
+  };
+
+  void addChannel( int uhtrIndex , edm::SortedCollection<HcalTriggerPrimitiveDigi>::const_iterator qiedf , int verbosity = 0 ){
+    // loop over words in dataframe
+    for( int iTS = 0 ; iTS < qiedf->size() ; iTS++ ){
+      // push data into uhtr data container
+      uhtrs[uhtrIndex].push_back(qiedf->sample(iTS).raw());
+    }// end loop over dataframe words
+  };
+
+  void addChannel( int uhtrIndex , QIE11DataFrame qiedf , int verbosity = 0 ){ 
+    // loop over words in dataframe 
+    for(edm::DataFrame::iterator dfi=qiedf.begin() ; dfi!=qiedf.end(); ++dfi){      
+      if(dfi == qiedf.begin()) ++dfi; // hack to prevent double channel header
+      // push data into uhtr data container
+      uhtrs[uhtrIndex].push_back(dfi[0]);
+    }// end loop over dataframe words
+  };
+
+  void addChannel( int uhtrIndex , QIE10DataFrame qiedf , int verbosity = 0 ){ 
+    // loop over words in dataframe 
+//    int dfcnt =-1;
+
+    for(edm::DataFrame::iterator dfi=qiedf.begin() ; dfi!=qiedf.end(); ++dfi){      
+//      if(dfi == qiedf.begin()) ++dfi; // hack to prevent double channel header
+//      dfcnt ++;
+//      if( dfi != qiedf.begin() && (((*dfi)&0x8000) !=0) ){
+//      if( dfi == qiedf.end()-1 ){
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant